modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ImagePicker/ImagePicker.es.js

```diff
@@ -12,160 +12,275 @@
-
-	/**
-	 * @default false
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(bool|undefined)}
-	 */
-
-	required: Config.bool().value(false),
-
-	/**
-	 * @default true
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(bool|undefined)}
-	 */
-
-	showLabel: Config.bool().value(true),
-
-	/**
-	 * @default undefined
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(string|undefined)}
-	 */
-
-	spritemap: Config.string(),
-
-	/**
-	 * @default undefined
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(string|undefined)}
-	 */
-
-	tip: Config.string(),
-
-	/**
-	 * @default undefined
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(string|undefined)}
-	 */
-
-	value: Config.string(),
+	};
+
+	const _handleItemSelectorTriggerClick = event => {
+		event.preventDefault();
+
+		const itemSelectorDialog = new ItemSelectorDialog({
+			eventName: `${portletNamespace}selectDocumentLibrary`,
+			singleSelect: true,
+			url: itemSelectorURL,
+		});
+
+		itemSelectorDialog.on('selectedItemChange', _handleFieldChanged);
+
+		itemSelectorDialog.open();
+	};
+
+	const spritemap =
+		Liferay.ThemeDisplay.getPathThemeImages() + '/lexicon/icons.svg';
+
+	const placeholder = readOnly
+		? ''
+		: Liferay.Language.get('add-image-description');
+
+	let {height = '0', width = '0'} = {
+		...JSON.parse(inputValue || '{}'),
+	};
+
+	height = `${height}px`;
+	width = `${width}px`;
+
+	return (
+		<>
+			<ClayForm.Group style={{marginBottom: '0.5rem'}}>
+				<input
+					id={id}
+					name={name}
+					type="hidden"
+					value={JSON.stringify(imageValues)}
+				/>
+				<ClayInput.Group>
+					<ClayInput.GroupItem className="d-none d-sm-block" prepend>
+						<ClayInput
+							className="bg-light"
+							disabled={readOnly}
+							onClick={_handleItemSelectorTriggerClick}
+							readOnly
+							type="text"
+							value={imageValues.title}
+						/>
+					</ClayInput.GroupItem>
+
+					<ClayInput.GroupItem append shrink>
+						<ClayButton
+							disabled={readOnly}
+							displayType="secondary"
+							onClick={_handleItemSelectorTriggerClick}
+							type="button"
+						>
+							{Liferay.Language.get('select')}
+						</ClayButton>
+					</ClayInput.GroupItem>
+
+					{imageValues.url && (
+						<ClayInput.GroupItem shrink>
+							<ClayButton
+								disabled={readOnly}
+								displayType="secondary"
+								onClick={_handleClearClick}
+								type="button"
+							>
+								{Liferay.Language.get('clear')}
+							</ClayButton>
+						</ClayInput.GroupItem>
+					)}
+				</ClayInput.Group>
+			</ClayForm.Group>
+
+			{imageValues.url && modalVisible && (
+				<ClayModal
+					className="image-picker-preview-modal"
+					observer={observer}
+					size="full-screen"
+					spritemap={spritemap}
+				>
+					<ClayModal.Header />
+					<ClayModal.Body>
+						<img
+							alt={imageValues.description}
+							className="d-block img-fluid mb-2 mx-auto rounded"
+							onClick={() => setModalVisible(false)}
```

Comment:
do not call `setModalVisible` directly, this will interrupt the animation when closing the modal. The `useModal` hook returns `onClose`, you can use it here. https://clayui.com/docs/components/modal.html#usemodal-hook

Reply:
Thanks for the tip! Implementing!